src/main/java/XbeLoader/XbeLoader.java

```diff
@@ -460,22 +460,36 @@ protected void load(ByteProvider provider, LoadSpec loadSpec, List<Option> optio
 
 		// Read XBE header
 		header = new XbeImageHeader(reader);
-
 		try {
 			program.setImageBase(api.toAddr(header.baseAddr), true);
 		} catch (Exception e) {
 			Msg.error(this, e.getMessage());
 		}
 
+		// Read headers into memory
+		reader.setPointerIndex(0);
+		createSection(api, "headers", reader,
+				header.baseAddr, header.headersSize,
+				0, header.headersSize, false, false);
+
+		long entry = header.entryAddr;
+
 		// Unscramble entry point
-		long ENTRY_DEBUG  = 0x94859D4BL;
-		long ENTRY_RETAIL = 0xA8FC57ABL;
-		long entry = header.entryAddr ^ ENTRY_DEBUG;
-		if (entry < 0x4000000) {
-			isDebug = true;
+		// https://github.com/radareorg/radare2/blob/0acfd3d3/libr/bin/p/bin_xbe.c#L33
+		if ((entry & 0xF0000000) == 0x40000000) {
+			// segaboot specific
+			entry ^= 0x40B5C16EL;
+			kernelThunkTableAddr = api.toAddr(header.kernThunkAddr ^ 0x2290059DL);
 		} else {
-			entry = header.entryAddr ^ ENTRY_RETAIL;
-			isDebug = false;
+			if ((entry ^ 0x94859D4BL) < 0x4000000L) {
```

Comment:
- Keep names for constants
- Simplify if else

Reply:
Pushed requested changes.